tracing-subscriber/src/layer/mod.rs

```diff
@@ -395,40 +395,80 @@
 //!
 //! ## Runtime Configuration With Layers
 //!
-//! Conditionally enabling/disabling layers at runtime can be challenging due to the generics
-//! used Layers implementations. However, A Layer wrapped in an `Option` is also Layers, which
-//! allows some runtime configuration:
+//! In some cases, a particular [`Layer`] may be enabled or disabled based on
+//! runtime configuration. This can introduce challenges, because the type of a
+//! layered [`Subscriber`] depends on which layers are added to it: if an `if`
+//! or `match` expression adds some [`Layer`]s in one branch and other layers
+//! in another, the [`Subscriber`] values returned by those branches will have
+//! different types. For example, the following _will not_ work:
+//!
+//! ```compile_fail
+//! # fn docs() -> Result<(), Box<dyn std::error::Error + 'static>> {
+//! # use std::path::PathBuf;
+//! # struct Config {
+//! #    is_prod: bool,
+//! #    path: PathBuf,
+//! # }
+//! # let cfg = Config { is_prod: false, path: PathBuf::from("debug.log") };
```

Comment:
nit/tioli: using `PathBuf` is probably more correct here, but since this is just an example, this code is hidden, and we don't actually care about _creating_ the log file, and never do any interesting path operations on it, we could simplify this a bit
```suggestion
//! # struct Config {
//! #    is_prod: bool,
//! #    path: &'static str,
//! # }
//! # let cfg = Config { is_prod: false, path: "debug.log" };
```